CMP/CmpWap/CmpWapExtension/Api/SyncWorker.cs

```diff
@@ -1,516 +1,516 @@
+                    Code = resGroup.Code,
+                    Name = resGroup.Name,
+                    HasService = (bool)resGroup.HasService,
+                    IsActive = resGroup.IsActive,
+                    SubscriptionId = resGroup.SubscriptionId,
+                    CreatedOn = resGroup.CreatedOn,
+                    CreatedBy = resGroup.CreatedBy,
+                    LastUpdatedOn = resGroup.LastUpdatedOn,
+                    LastUpdatedBy = resGroup.LastUpdatedBy,
+                    CIOwner = resGroup.CIOwner,
+                    Region = resGroup.Region
+                };
+
+                cwdb.InsertApp(app);
+            }
+            catch (Exception ex)
+            {
+                LogThis(ex, EventLogEntryType.Error,
+                    "Exception in SyncWorker.ImportApp()", 0, 0);
+            }
+        }
+
+        //*********************************************************************
+        ///
+        /// <summary>
+        /// Sync the CMP VM list with the CmpWap VM list
+        /// </summary>
+        /// 
+        //*********************************************************************
+
+        public void SynchWithCmp()
+        {
+            var cmp = new CmpApiClient(_eventLog);
+            var cwdb = new CmpWapDb();
+
+            //*** Fetch the CMP resource group list ***
+            var cmpResGroupList = cmp.FetchAzureResourceGroups();
+
+            //*** Fetch the CmpWap app list ***
+            var cmpWapAppList = cwdb.FetchAppList();
+
+            //*** Fold ***
+            foreach (var cmpResGroup in cmpResGroupList.Where(cmpResGroup =>
+                !cmpWapAppList.Any(cmpWapApp => cmpResGroup.Name.Equals(
+                    cmpWapApp.Name, StringComparison.InvariantCultureIgnoreCase))))
+                ImportApp(cmpResGroup, cwdb);
+
+            //*** Fetch the CMP VM list ***
+            var cmpVmList = cmp.FetchCmpRequests();
+
+            //*** Fetch the CmpWap VM list ***
+            var cmpWapVmList = cwdb.FetchVmDepRequests(null, true);
+
+            //*** Fold ***
+            foreach (var cmpVm in cmpVmList.Where(cmpVm =>
+                !cmpWapVmList.Any(cmpWapVm => cmpVm.TargetVmName.Equals(
+                    cmpWapVm.TargetVmName, StringComparison.InvariantCultureIgnoreCase))))
+                ImportVm(cmpVm, cwdb);
+        }
+
+        #endregion
+
+        #region --- Azure Sync Worker --
+
+        private void AzureWorker()
+        {
+            var retryCount = 1;
+
+            while (true)
+            {
+                try
+                {
+                    SyncWithAzure();
+                    Thread.Sleep(AzureSleepTime);
+                }
+                catch (Exception ex)
+                {
+                    var error = Utilities.UnwindExceptionMessages(ex);
+
+                    if (error.Contains("503") & (5 > retryCount++))
+                        Thread.Sleep((int)(1000 * Math.Pow(4.17, retryCount)));

```

Comment:
Just read your notes on the email. If it requires re-installation, we need to put it on the release notes, like last time. Maybe a work around for people that do not want to re-install. (I'm assuming it's just adding this config key and restarting CMP)